### ISSUE TITLE
[css-images] image-set() resolution should be applied on-top of, not instead-of the natural image resolution.

### ISSUE THREAD

**Issue by emilio** [w3c/csswg-drafts#6241]
Right now https://drafts.csswg.org/css-images-4/#image-set-notation says:

> If not specified, it behaves as 1x for the purpose of selecting which <image-set-option> to use. It also defaults the image’s natural resolution to 1x, but if some other source of data supplies a natural resolution, that resolution must be honored instead.

That doesn't seem to match implementations, which apply the scaling on top of the natural orientation of the image. See https://github.com/whatwg/html/pull/5574#issuecomment-826347560 and previous discussion.

cc @noamr

**Comment by zcorpan** [comment]
So if I understand correctly, we want this:

- First. apply the EXIF intrinsic sizing
- Then, apply the image-set() or srcset density as a multiplier.

**Comment by noamr** [comment]
> So if I understand correctly, we want this:
> 
> * First, apply the EXIF intrinsic sizing
> * Then, apply the image-set() or srcset density as a multiplier.

Exactly. That’s the current implementation. If WPT/selenium supported DPR we could add tests for this.

**Comment by zcorpan** [comment]
OK, thanks. It should work that way both for same-origin and CORS-cross-origin, right?

**Comment by zcorpan** [comment]
If so, the `from-image` value is equivalent to `1x`, right? Should we drop `from-image`?

**Comment by noamr** [comment]
> OK, thanks. It should work that way both for same-origin and CORS-cross-origin, right?

Yes, origin is not playing a part in this. The rationale was we didn't want the image's CORS-ness to affect its natural size in implicit ways. 

Where CORS would play a part would be if we ever implement CSS `image-resolution` in a way that would override the EXIF resolution behavior.

**Comment by noamr** [comment]
> If so, the `from-image` value is equivalent to `1x`, right? Should we drop `from-image`?

You mean `image-resolution: from-image`? See above. If we were to implement this, it is where CORS would kick in - as `image-resolution` would expose whether an image is EXIF-resolution-corrected and that information shouldn't leak cross origins.

I am not sure there is an actual use for `image-resolution` - I think `image-orientation` is used mainly for backwards compatibility issues, which is less of a problem with resolution.

**Comment by zcorpan** [comment]
> You mean `image-resolution: from-image`?

Yes.

> See above. If we were to implement this, it is where CORS would kick in - as `image-resolution` would expose whether an image is EXIF-resolution-corrected and that information shouldn't leak cross origins.
> 
> I am not sure there is an actual use for `image-resolution` - I think `image-orientation` is used mainly for backwards compatibility issues, which is less of a problem with resolution.

OK. I'll file a separate issue about `image-resolution` since this issue is about `image-set()`. #6317

**Comment by zcorpan** [comment]
> If WPT/selenium supported DPR we could add tests for this.

I think it should be testable anyway. Have an image with 2x density per EXIF, and use it in `image-set("image.png" 3x)`. Test that the image has 6x density with `.clientWidth`.
